File: app/src/main/java/com/example/dcct/base/BasePresenter.java
package com.example.dcct.base;

import java.lang.ref.WeakReference;

public class BasePresenter<T> {
    protected WeakReference<T> mCallBackView;

    public void attachView(T view){
        mCallBackView = new WeakReference<>( view );//弱引用
    }

    public void detachView(){
        if (mCallBackView != null) {
            mCallBackView.clear();
            mCallBackView = null;
        }
    }
}

File: app/src/main/java/com/example/dcct/base/BasePresenter.kt
package com.example.dcct.base

import java.lang.ref.WeakReference

open class BasePresenter<T> {
    @JvmField
    protected var mCallBackView: WeakReference<T>? = null
    fun attachView(view: T) {
        mCallBackView = WeakReference(view) //弱引用
    }

    fun detachView() {
        if (mCallBackView != null) {
            mCallBackView!!.clear()
            mCallBackView = null
        }
    }
}